pkg/util/fieldindex/register.go

```diff
@@ -71,6 +64,79 @@ func RegisterFieldIndexes(c cache.Cache) error {
 		if err = c.IndexField(&v1.PersistentVolumeClaim{}, IndexNameForOwnerRefUID, ownerIndexFunc); err != nil {
 			return
 		}
+		// pod name
+		if err = indexPodName(c); err != nil {
+			return
+		}
+		// job owner
+		if err = indexJob(c); err != nil {
+			return
+		}
+		// broadcastjob owner
+		if err = indexBroadcastCronJob(c); err != nil {
+			return
+		}
 	})
 	return err
 }
+
+func indexPodName(c cache.Cache) error {
+	if err := c.IndexField(&v1.Pod{}, IndexNameForPodNodeName, func(obj runtime.Object) []string {
+		pod, ok := obj.(*v1.Pod)
+		if !ok {
+			return []string{}
+		}
+		if len(pod.Spec.NodeName) == 0 {
+			return []string{}
+		}
+		return []string{pod.Spec.NodeName}
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func indexJob(c cache.Cache) error {
+	if err := c.IndexField(&batchv1.Job{}, IndexNameForController, func(rawObj runtime.Object) []string {
+		// grab the job object, extract the owner...
+		job := rawObj.(*batchv1.Job)
+		owner := metav1.GetControllerOf(job)
+		if owner == nil {
+			return nil
+		}
+
+		// ...make sure it's a AdvancedCronJob...
+		if owner.APIVersion != apiGVStr || owner.Kind != appsv1alpha1.AdvancedCronJobKind {
+			return nil
+		}
+
+		// ...and if so, return it
+		return []string{owner.Name}
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func indexBroadcastCronJob(c cache.Cache) error {
```

Comment:
how about change the style just like:
```go
return  c.IndexField(&appsv1alpha1.BroadcastJob{}, IndexNameForController, func(rawObj runtime.Object) []string {
		// grab the job object, extract the owner...
		job := rawObj.(*appsv1alpha1.BroadcastJob)
		owner := metav1.GetControllerOf(job)
		if owner == nil {
			return nil
		}

		// ...make sure it's a AdvancedCronJob...
		if owner.APIVersion != apiGVStr || owner.Kind != appsv1alpha1.AdvancedCronJobKind {
			return nil
		}

		// ...and if so, return it
		return []string{owner.Name}
	})
```

Reply:
Done in commit https://github.com/openkruise/kruise/pull/461/commits/3d513814466ce219a7543280c4efc3a697a9fe78